frame/support/src/storage/bounded_vec.rs

```diff
@@ -192,6 +197,13 @@ impl<T, S: Get<u32>> BoundedVec<T, S> {
 			Err(())
 		}
 	}
+
+	/// Create a new bounded vec from an arbitrary sized vec by taking the first
+	/// items up till the bound.
+	pub fn truncating_from(i: Vec<T>) -> Self {
+		let bounded = i.into_iter().take(S::get() as usize).collect();
+		Self::unchecked_from(bounded)
+	}
```

Comment:
is this the idiomatic way to do it?

Reply:
Well, you can observe from your usage that you don't really care whether the `i` is a `Vec` or not; all you really care in fact is whether you can create an iterator of `T` elements out of it. So, what you could do instead is accept a type that implements `Iterator<Item = T>`.